tensorflow/lite/examples/label_image/CMakeLists.txt

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# The label_image example for Tensorflow Lite.
+
+populate_source_vars("${TFLITE_SOURCE_DIR}/examples/label_image"
+  TFLITE_LABEL_IMAGE_SRCS
+  FILTER "_test\\.cc$"
+)
+list(APPEND TFLITE_LABEL_IMAGE_SRCS
+  ${TF_SOURCE_DIR}/core/util/stats_calculator.cc
+  ${TFLITE_SOURCE_DIR}/profiling/memory_info.cc
+  ${TFLITE_SOURCE_DIR}/profiling/profile_summarizer.cc
+  ${TFLITE_SOURCE_DIR}/profiling/profile_summary_formatter.cc
+  ${TFLITE_SOURCE_DIR}/profiling/time.cc
+  ${TFLITE_SOURCE_DIR}/tools/command_line_flags.cc
+  ${TFLITE_SOURCE_DIR}/tools/delegates/default_execution_provider.cc
+  ${TFLITE_SOURCE_DIR}/tools/evaluation/utils.cc
+  ${TFLITE_SOURCE_DIR}/tools/tool_params.cc
+)
+
+if(TFLITE_ENABLE_XNNPACK)
```

Comment:
did you test these delegates?

Reply:
@terryheo
Yes, I add another commit(https://github.com/tensorflow/tensorflow/pull/47317/commits/79e6e84b602b7a42ddb8aaa589f16367a419b5c3) to show the delegation log.

I add the addition def "TFLITE_PROFILING_ENABLED" to show the profiling result in label_image.
```
diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index 74f1dc8..9dbfb0f 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -143,7 +143,7 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
   list(APPEND TF_TARGET_PRIVATE_OPTIONS "-Wno-deprecated-declarations")
 endif()
 # Additional compiler flags used when compiling TF Lite.
-set(TFLITE_TARGET_PUBLIC_OPTIONS "")
+set(TFLITE_TARGET_PUBLIC_OPTIONS "-DTFLITE_PROFILING_ENABLED")
 set(TFLITE_TARGET_PRIVATE_OPTIONS "")
 # Additional library dependencies based upon enabled features.
 set(TFLITE_TARGET_DEPENDENCIES "")	
```

The log:
```
/tensorflow/tensorflow/lite/build$ ./examples/label_image/label_image -m ~/mobilenet_v1_1.0_224.f32.tflite -l ~/labels.txt -i /tensorflow/tensorflow/lite/examples/label_image/testdata/grace_hopper.bmp -p 1 -x 1

INFO: Loaded model /home/jerrys/mobilenet_v1_1.0_224.f32.tflite
INFO: resolved reporter
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
INFO: Use XNNPACK acceleration.
INFO: Applied XNNPACK delegate.
ERROR: ModifyGraphWithDelegate is disallowed when graph is immutable.
ERROR: Ignoring failed application of the default TensorFlow Lite delegate indexed at 0.
INFO: invoked
INFO: average time: 6.421 ms
INFO:      6.411, Subgraph   0, Node   0, OpCode   3, CONV_2D
INFO:      6.388, Subgraph   0, Node  31, OpCode  51, DELEGATE
INFO:      0.002, Subgraph   0, Node  29, OpCode  43, SQUEEZE
INFO:      0.008, Subgraph   0, Node  32, OpCode  51, DELEGATE
INFO: 0.860174: 653 653:military uniform
INFO: 0.0481019: 907 907:Windsor tie
INFO: 0.00786707: 466 466:bulletproof vest
INFO: 0.00644937: 514 514:cornet, horn, trumpet, trump
INFO: 0.00608026: 543 543:drumstick
```

You could see the message:
```
INFO: Applied XNNPACK delegate.
ERROR: ModifyGraphWithDelegate is disallowed when graph is immutable.
ERROR: Ignoring failed application of the default TensorFlow Lite delegate indexed at 0.
```

The xnnpack backend was created successfully.

The "ModifyGraphWithDelegate is disallowed when graph is immutable" is another issue:https://github.com/tensorflow/tensorflow/issues/42757

Reply:
The "DELEGATE" op in profiling result shows that the graph goes through delegation backend.